src/ui/components/CredentialDetailModule/CredentialDetailModule.types.ts

```diff
@@ -13,9 +14,11 @@ enum BackReason {
 interface CredentialDetailModuleBaseProps {
   pageId: string;
   id: string;
+  credDetail?: ACDCDetails;
   onClose?: (reason: BackReason) => void;
   navAnimation?: boolean;
   hardwareBackButtonConfig?: HardwareBackButtonConfig;
+  viewOnly?: boolean;
```

Comment:
Is credDetail (maybe should be `credDetails`) always passed when `viewOnly` is true?

If so, it's better to have something like:
```
type CredentialDetailModuleBaseProps = {
  pageId: string;
  id: string;
  onClose?: (reason: BackReason) => void;
  navAnimation?: boolean;
  hardwareBackButtonConfig?: HardwareBackButtonConfig;
} & { viewOnly: false } | { viewOnly: true; credDetails: ACDCDetails; };
```
(this might be slightly off, I haven't tested)

So if `viewOnly` is true, you must pass `credDetails`.

But if not, it's OK as you have it.

Reply:
reverted since the 2 props are not tightly bound